### ISSUE TITLE
ref.on('value', callback) not calling callback sometimes

### ISSUE THREAD

**Issue by graphee-gabriel** [invertase/react-native-firebase#943]
### Issue

Hello
I'm having the weirdest issue:
I have a list of component showing some data (title + description)
Each component has a different database ref with a listener .on('value')
One listener on the name's ref, one on the description's ref.
Those listener update the state to show the name and description.
If I click on it an item, i replace this list of item by the detail view of an item, that has the same component with title and description, and some other things.
But here when I remove the list of item, and mount instead this one item, the callback from ref.on('value, callback) is never fired. The detail view opens on an empty name and description.

I logged everything, and the function .on('value', callback) is called, just the callback never fires again.
The only way to get it to fire again, is to go back to the list, select another item (who's callback will never fire again) and come back to the list.

Basically: view list [A, B, C] -> everything is ok
Click on A -> callback not firing === no name no description
Go Back to list -> A lost its name and description, the callback is not firing here either.
Click on B -> callback not firing === no name no description for B
Go back to list -> B lost its name and description, but A is back on

I'm really puzzled...

### Environment

iOS

1. Application Target Platform:

8.0

2. Development Operating System:

OSX, Xcode Version 9.3 (9E145)

3. Build Tools:

?

4. React Native version:

0.55.1

5. RNFirebase Version:

4.0.0

6. Firebase Module:

Database

**Comment by graphee-gabriel** [comment]
It seems that having this component "re-used" by react natives conflicts with the componentWillUnmount, calling the .off() on the database ref... which seems to be the reason why the callback is not called anymore. I'm guessing then this is not a react-native-firebase issue.

**Comment by chrisbianca** [comment]
@graphee-gabriel it sounds like you're removing the subscription for all listeners, not just the one relevant to your specific component.  `ref.on()` returns a function which you can pass into `ref.off()` to remove the specific listener rather than any listener on the particular reference.
